pkg/components/dex/component.go

```diff
@@ -190,6 +190,7 @@ metadata:
     cert-manager.io/cluster-issuer: {{ .CertManagerClusterIssuer }}
     meta.helm.sh/release-name: dex
     meta.helm.sh/release-namespace: dex
+    external-dns.alpha.kubernetes.io/hostname: {{ .IngressHost }}
```

Comment:
This annotation is optional for external-dns. It should pick up the hostname from the Ingress object itself. I think we can remove it.

Reply:
Also, this annotation is for objects `kind: Service`, not `kind: Ingress` as far as I know. We can definitely remove it.